src/core/diags/README.md

```diff
@@ -337,6 +338,29 @@ stop diagnostics mode
 status 0x00
 ```
 
+### diag radiospinel
```

Comment:
These diag commands are only available when the radio spinel is used. We may need to add an extra `DIAG_RADIO_SPINEL_README.md` to `src/lib/spinel`  to describe these commands. Thoughts? @abtink

Reply:
Created `README_RADIO_SPINEL_DIAG.md` based on https://github.com/openthread/openthread/blob/main/src/posix/platform/README_RCP_CAPS_DIAG.md